pages/pil/training/routers/modules.js

```diff
@@ -14,13 +14,35 @@ module.exports = settings => {
     next();
   });
 
-  app.use('/', form({ schema }));
+  app.use('/', form({
+    schema: {
+      ...schema,
+      ...schema.modules.options.reduce((obj, val) => {
+        return {
+          ...obj,
+          [`module-${val.value}-species`]: val.species
+
+        };
+      }, {})
+    },
+    locals: (req, res, next) => {
+      res.locals.static.schema = schema;
+      next();
+    }
+  }));
 
   app.post('/', (req, res, next) => {
     const fields = ['certificateNumber', 'accreditingBody', 'passDate', 'modules'];
     const values = pick(req.session.form[req.model.id].values, fields);
 
-    values.modules = values.modules.map(module => ({ module, species: [] }));
+    values.modules = values.modules.map(module => {
+      const species = req.form.values[`module-${module}-species`];
+      return { module,
+        species:
+         Array.isArray(species) ? species.filter(s => s !== '') : (species !== '' ? [species] : [])
```

Comment:
indentation

Reply:
also, what are you trying to do here? It looks a lot like you want `castArray` or something

Reply:
See my other comment 👇